app/cdap/components/GridTable/index.tsx

```diff
@@ -208,59 +234,73 @@ export default function GridTable({ handleTransformationUpload, storeData }) {
-                          />
-                        );
-                      })}
-                    </TableRow>
-                  );
-                })}
-              </TableBody>
-            </Table>
-          </TableWrapper>
-        ) : (
+      <GridWrapper data-testid="grid-table-container">
+        {!dataprep.headers.length && (
           <FlexWrapper>
             <NoRecordScreen
               title={T.translate('features.WranglerNewUI.NoRecordScreen.gridTable.title')}
               subtitle={T.translate('features.WranglerNewUI.NoRecordScreen.gridTable.subtitle')}
             />
           </FlexWrapper>
         )}
-        <FooterPanel recipeStepsCount={0} gridMetaInfo={tableMetaInfo} />
+
+        {dataprep.headers.length && (
+          <TableGridContainer>
+            <TableWrapperContainer>
+              <Table aria-label="simple table" className="test">
+                <TableHead>
+                  <TableRow>
+                    {dataprep.headers.map((eachHeader) => (
+                      <GridHeaderCell
+                        label={eachHeader}
+                        type={dataprep.types[eachHeader]}
+                        key={eachHeader}
+                        columnSelected={selectedColumn}
+                        setColumnSelected={handleColumnSelect}
+                      />
+                    ))}
+                  </TableRow>
+                  <TableRow>
+                    {dataprep.headers.map((each) => {
+                      return createMissingData(columnsInformation.columns).map((item) => {
+                        if (item.name === each) {
+                          return <GridKPICell metricData={item} key={item.name} />;
+                        }
+                      });
+                    })}
+                  </TableRow>
+                </TableHead>
+                <TableBody>
+                  {dataprep.data.map((eachRow, rowIndex) => {
+                    return (
+                      <TableRow key={`row-${rowIndex}`}>
+                        {dataprep.headers.map((eachKey, eachIndex) => {
+                          return (
+                            <GridTextCell
+                              cellValue={eachRow[eachKey] || '--'}
+                              key={`${eachKey}-${eachIndex}`}
+                            />
+                          );
+                        })}
+                      </TableRow>
+                    );
+                  })}
+                </TableBody>
+              </Table>
+            </TableWrapperContainer>
+            {showRecipePanel && (
+              <RecipeStepsPanel
+                onDrawerCloseIconClick={() => setShowRecipePanel(false)}
+                setSnackbar={setSnackbar}
+              />
+            )}
+          </TableGridContainer>
+        )}
+
+        <FooterPanel
+          gridMetaInfo={tableMetaInfo}
+          toggleRecipePanelMode={() => setShowRecipePanel(!showRecipePanel)}
```

Comment:
change toggleRecipePanelMode to onRecipeStepsButtonClick

Reply:
changed